examples/arithmetics/src/language-server/arithmetics-util.ts

```diff
@@ -4,9 +4,13 @@
  * terms of the MIT License, which is available in the project root.
  ******************************************************************************/
 
-export function applyOp(op: '+' | '-' | '*' | '/'): (x: number, y: number) => number {
-    if (op === '+') return (x, y) => x + y;
-    if (op === '-') return (x, y) => x - y;
-    if (op === '*') return (x, y) => x * y;
-    return (x, y) => y === 0 ? x : x / y;
+export function applyOp(op: '+' | '-' | '*' | '/' | '^' | '%'): (x: number, y: number) => number {
+    switch (op) {
+        case '+': return (x, y) => x + y;
+        case '-': return (x, y) => x - y;
+        case '*': return (x, y) => x * y;
+        case '^': return (x, y) => Math.pow(x, y);
+        case '%': return (x, y) => x % y;
+        default: return (x, y) => y === 0 ? x : x / y;
```

Comment:
In addition to what @sailingKieler already said about this: It's fine to throw an error here when we try to divide by 0. I know it's old code, but we should catch it appropriately. Simply returning `x` is very much unexpected in this instance.